curriculum/challenges/english/25-front-end-development/lecture-working-with-loops-and-sequences/6839e3ea328b630cf5fca9ac.md

```diff
@@ -1,131 +1,293 @@
 id: 6839e3ea328b630cf5fca9ac
 title: How Do Loops Work?
-challengeType: 11
-videoId: nVAaxZ34khk
+challengeType: 19
 dashedName: how-do-loops-work
 ---
 
-# --description--
+# --transcript--
 
-Watch the video or read the transcript and answer the questions below.
+As you learned in earlier modules, loops are used to repeat a block of code for a set number of times. In this lecture you will learn how to work with different types of loops in Python.
+
+The first loop we will go over is the `for` loop. Here is an example of using a `for` loop to iterate through a list and print each item to the console:
+
+```py
+programming_languages = ['Rust', 'Java', 'Python', 'C++']
+
+for language in programming_languages:
+    print(language)
+
+"""
+Result:
+
+'Rust'
+'Java'
+'Python'
+'C++'
+"""
+```
+
+Notice that the `print(language)` is indented inside of the loop. Without that indentation, you would get an `IndentationError`:
+
+```python
+"""
+Traceback (most recent call last):
+  File "<stdin>", line 2, in <module>
+IndentationError: expected an indented block after 'for' statement on line 1
+"""
+```
+
+You can also use a `for` loop to iterate through other iterables like a string. Here is an example of using a `for` loop to loop through the string `code` and print out each character:
+
+```py
+for char in 'code':
+    print(char)
+
+"""
+Result:
+
+'c'
+'o'
+'d'
+'e'
+"""
+```
+
+Just like in JavaScript, you can also nest `for` loops in Python. Here is an example of using a nested `for` loop:
+
+```py
+categories = ['Fruit', 'Vegetable']
+foods = ['Apple', 'Carrot', 'Banana']
+
+for category in categories:
+    for food in foods:
+        print(category, food)
+```
+
+The outer loop will iterate through each `category` in the `categories` list. For each `category`, the inner loop will iterate through each `food` in the `foods` list. Here is the result that will be printed to the console:
+
+```md
+Fruit Apple
+Fruit Carrot
+Fruit Banana
+Vegetable Apple
+Vegetable Carrot
+Vegetable Banana
+```
+
+Another type of loop you can use in Python is the `while` loop. This type of loop will repeat a block of code until the condition is `False`. Here is an example of using a `while` loop for a guessing game:
+
+```py
+secret_number = 3
+guess = 0
+
+while guess != secret_number:
+    guess = int(input('Guess the number (1-5): '))
+    if guess != secret_number:
+        print('Wrong! Try again.')
+
+print('You got it!')
+```
+
+In this example we have a `secret_number` variable with the value of `3` and an initial guess of `0`. Then we use the `input` function to get input from the user, then convert the input string into and integer with the `int()` function, and assign it to the `guess` variable. If the user guesses correctly by inputting `3`, the `while` loop is broken out of and the message `You got it!` is printed to the console. Otherwise, the message `Wrong! Try again` is printed to the console, and the loop repeats, prompting the user to guess again.
```

Comment:
```suggestion
In this example we have a `secret_number` variable with the value of `3` and an initial guess of `0`. Then we use the `input` function to get input from the user, then convert the input string into and integer with the `int()` function, and assign it to the `guess` variable. If the user guesses correctly by inputting `3`, the `while` loop is broken out of and the message `You got it!` is printed to the console. Otherwise, the message `Wrong! Try again.` is printed to the console, and the loop repeats, prompting the user to guess again.
```